transform/all_collection/spark/process.py

```diff
@@ -0,0 +1,125 @@
+# pylint: disable=line-too-long, wildcard-import, invalid-name, consider-using-with, unused-wildcard-import
+"""Load, transform and send data"""
+import os
+import traceback
+from typing import Dict
+import requests
+from tqdm import tqdm
+from pyspark.sql import SparkSession
+from transform.all_collection.spark.conf.spark import get_spark_app_config
+from transform.all_collection.spark.conf.logger import Log4J
+from transform.all_collection.spark.utils.join_dfs import join_identical_dfs
+from transform.all_collection.spark.utils.loader import *
+from transform.all_collection.spark.utils.utils import check_dfs_cols
+from transform.all_collection.spark.utils.save_df import save_df
+from transform.all_collection.spark.schemas.harvested_props_schemas import (
+    harvested_schemas,
+)
+
+
+def check_trans_consistency(collections: Dict) -> None:
+    """Check whether all data types after transformations will have the schema
+    such that it will be possible to send all data into single solr collection"""
+    dfs = {}
+    for name, _col in collections.items():
```

Comment:
Why are you using two different cases while unpacking? 

`name`

`_col`, with an underscore

Reply:
`_col`  - I used here an underscore because col is a method in some packages like `pyspark,` `pandas` etc. So it potentially may shadow variable from outer scope.

I have changed those variables to:
`name` -> `collection_name` (col_name is in outer scope so it should not be used here)
`_col` -> `collection`